v2/service/module/tracerprovider/tracerprovider.go

```diff
@@ -128,29 +179,66 @@ func WithSamplePercentage(percentage int) Opt {
 	}
 }
 
+// WithCollector is deprecated. Use WithGRPCExporter instead.
 func WithCollector(host string, port int, credentials credentials.TransportCredentials) Opt {
+	slog.Warn("WithCollector is deprecated, use WithGRPCExporter instead")
+	return WithGRPCExporter(fmt.Sprintf("%s:%d", host, port), credentials)
+}
+
+// WithGRPCExporter sets gRPC collector endpoint for trace exporter.
+// Endpoint is "host:port" of the collector and you can set transport credentials to "insecure.NewCredentials()"
+// if secure connection is not required.
+func WithGRPCExporter(endpoint string, credentials credentials.TransportCredentials) Opt {
+	return func(tp *TracerProvider) error {
+		tp.grpc.endpoint = endpoint
+		tp.grpc.credentials = credentials
+		return nil
+	}
+}
+
+// WithHTTPExporter sets HTTP exporter endpoint for trace exporter with an authorization token.
+// Endpoint is "host:port" of to the collector and you can set token as "ApiKey <token>" or "Bearer <token>".
+// Set token to empty if not needed. Insecure disables TLS.
+func WithHTTPExporter(endpoint string, token string, insecure bool) Opt {
 	return func(tp *TracerProvider) error {
-		tp.collectorHost = host
-		tp.collectorPort = port
-		tp.credentials = credentials
+		if token != "" && !(strings.HasPrefix(strings.ToLower(token), "apikey ") || strings.HasPrefix(strings.ToLower(token), "bearer ")) {
+			return ErrInvalidToken
```

Comment:
just thinking is it valid to return error if someone have prefix outside of apikey/bearer? Or should we print warning to logs instead?

Reply:
Are there actually something else? E.g. Elastic APM supports only these two: https://www.elastic.co/guide/en/observability/current/apm-open-telemetry-direct.html#apm-connect-open-telemetry-collector

Reply:
well, lets start with that